core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -114,42 +115,88 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
         m_granularity_mask(0) {}
 
   /** \brief  Total range */
+#if !defined(KOKKOS_ENABLE_DEPRECATED_CODE_4) || \
+    defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
```

Comment:
Whether or not we disable deprecation warnings shouldn't affect the interface exposed. Whether we warn or error out is determined in the implementation of `check_conversion_safety` anyway and there the guards make sense.
```suggestion
#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_4
```
What happens if `is_convertible` is false anyway? Wouldn't that already result in a compilation error? In that case, we can just disable the new interface unconditionally anyway, right?

Reply:
Updated my comment.

Reply:
> Whether we warn or error out is determined in the implementation of `check_conversion_safety` anyway and there the guards make sense.
> What happens if `is_convertible` is false anyway? Wouldn't that already result in a compilation error? In that case, we can just enable the new interface unconditionally anyway, right?

Yes, since the old interface simply accepted arguments that are implicitly convertible to the `member_type`, semantically it should be no different from enforcing the explicit `is_convertible` check to the interface. I placed in those ifdef guards as a form of compatibility safety and mainly to follow our regular process for changing user facing interface.
But I am for using the new interface right away and leave the ifdef gurads only in `check_conversion_safety` as well.

Reply:
> >Whether or not we disable deprecation warnings shouldn't affect the interface exposed.

> I agree with this statement in general. But the change you suggest would inhibit diagnosing narrowing conversions. (I am not saying we cannot come up with something else, just that the suggestion does not work.)

@dalg24, any opinion on replacing the old interface with the new interface right away and leave the two ifdef guards only in `check_conversion_saftey`?

Reply:
> @dalg24, any opinion on replacing the old interface with the new interface right away and leave the two ifdef guards only in check_conversion_saftey?

OK with it.

Reply:
Switched to use the new interfaces and removed ifdef guards. [3065552](https://github.com/kokkos/kokkos/pull/6754/commits/3065552efd27d9a9782707b8804b394f2af2bbdc).